src/lib/__tests__/credentials.test.ts

```diff
@@ -13,10 +13,11 @@ import {
   SubjectOutboundTransporter,
   waitForCredentialRecord,
 } from './helpers';
-import { CredentialRecord } from '../storage/CredentialRecord';
-import { CredentialPreview, CredentialPreviewAttribute, CredentialState } from '../modules/credentials';
+import { CredentialRecord } from '../modules/credentials';
+import { CredentialPreview, CredentialPreviewAttribute } from '../modules/credentials';
 import { InitConfig } from '../types';
 import logger from '../logger';
+import { CredentialState } from '../modules/credentials/CredentialState';
```

Comment:
Yes. 

Where do you think we should put `CredentialState`? `models` or `repository`? I would perhaps put it into `repository` because it's highly related to a `Record` and persisted as part of it. Models are not persisted except for `Did` in connections.

Reply:
I would say `models`. I agree it is highly related to the repository, but it is to track protocol state, which is broader than the repository. Or just leave it in the root